internal/ethapi/api.go

```diff
@@ -861,7 +861,14 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 	if evm.Cancelled() {
 		return nil, fmt.Errorf("execution aborted (timeout = %v)", timeout)
 	}
-	return result, err
+	// If the result contains a revert reason, try to unpack and return it.
+	if res.Err != nil && len(res.Revert()) > 0 {
+		reason, err := abi.UnpackRevert(res.Revert())
+		if err == nil {
+			return nil, fmt.Errorf("execution reverted: %v", reason)
+		}
+	}
+	return result, res.Err
```

Comment:
Sorry Marius, I guess your original approach is correct. The error returned here shouldn't include any internal error. We may need to parse it outside(e.g. Call).

Especially the current code will break `EstimateGas` which has the assumption that the error returned here is non-internal-evm error

Reply:
Ok, parsed it now outside of `EstimateGas`